src/app/tab3/login/login.component.ts

```diff
@@ -7,8 +10,34 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LoginComponent implements OnInit {
 
-  constructor() { }
+  loginForm: FormGroup;
+  
+  constructor(private formBuilder: FormBuilder,
+    private commonRequestService: CommonRequestService) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.initLoginForm();
+  }
 
+  initLoginForm() {
+    this.loginForm = this.formBuilder.group({
+      username: ['', Validators.compose([Validators.required, Validators.pattern("[a-zA-Z0-9.-_]{1,}@[a-zA-Z.-]{2,}[.]{1}[a-zA-Z]{2,}")])],
```

Comment:
Move this Validations into Validation Patterns.ts file and use from there.

Reply:
Done